pkg/providers/v1/aws.go

```diff
@@ -4371,8 +4375,24 @@ func (c *Cloud) GetLoadBalancer(ctx context.Context, clusterName string, service
 
 // GetLoadBalancerName is an implementation of LoadBalancer.GetLoadBalancerName
 func (c *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	// TODO: replace DefaultLoadBalancerName to generate more meaningful loadbalancer names.
-	return cloudprovider.DefaultLoadBalancerName(service)
+	if val := service.Annotations[ServiceAnnotationLoadBalancerName]; val != nil {
+		r := regexp.MustCompile(`^[a-zA-Z0-9-]{0,32}`)
```

Comment:
Instead of compiling the regex every time that `*Cloud.GetLoadBalancerName()` is called, consider placing this as a package-level var at the top of the file:

```go
var lbNameRegex = regexp.MustCompile(`^[a-zA-Z0-9-]{0,32}`)
```

and simply refer to it here...

```go
if lbNameRegex.MatchString(val) ...
```

Reply:
Update the regex as a package-level var.